src/com/jalasoft/search_files/utils/Search_Validation.py

```diff
@@ -9,36 +9,36 @@ def __init__(self):
     def string1_exists_within_string2(self, string_1, string_2, flag):
         case_sensitive = bool(flag)
         if case_sensitive is True:
-            search_match = re.search(string_1, string_2, re.IGNORECASE)
+            search_match = re.search(string_1, string_2)
```

Comment:
I could not find my last observation due to your first functions related to **"within"** and **"equal to"**. They should have different behaviors however they call to same function **"re.search"**
 I would like to share with you some cases that I researched to clarify which are the goals:
import re
# exact + ignote case
print(re.match('Maria', 'MARIA', re.IGNORECASE)) # true
print(re.match('Maria', 'aaaaMARIAaaa', re.IGNORECASE)) # False
# exact + non ignote case
print(re.match('Maria', 'Maria')) # true
print(re.match('Maria', 'aMariaa')) # False
# contain + ignote case
print(re.search('Maria', 'aaa aMARIAaa  aa', re.IGNORECASE)) # true
print(re.search('pedro', 'Sara', re.IGNORECASE)) # False
# contain + non ignote case
print(re.match('Maria', 'aaaaaMARIAaaaaa')) #false
print(re.search('Maria', 'aaaa Maria aaaaa')) # true
If you have any question, please, let me know